cpp/src/config.cpp

```diff
@@ -17,6 +20,59 @@
 #include <rapidsmpf/config.hpp>
 #include <rapidsmpf/utils.hpp>
 
+namespace {
+// Serialization limits and format configuration (implementation details)
```

Comment:
Nit (throughout): if these are supposed to be byte streams, let us use `std::byte` for them.